### PR TITLE
MCR-2281 add functions for i18n

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2281).

### CODE REVIEW

File: mycore-base/src/main/resources/xsl/functions/i18n.xsl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:xlink="http://www.w3.org/1999/xlink"
+                xmlns:fn="http://www.w3.org/2005/xpath-functions"
+                xmlns:map="http://www.w3.org/2005/xpath-functions/map"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:mcri18n="http://www.mycore.de/xslt/i18n"
+                exclude-result-prefixes="fn xs">
+
+    <xsl:function name="mcri18n:translate" as="xs:string">
+        <xsl:param name="key" as="xs:string"/>
```
Comment (yagee-de) [inline mycore-base/src/main/resources/xsl/functions/i18n.xsl:12]: we should (at least as a dummy support) a 2nd parameter "values" that is a sequence of string values that can be filled in gaps in the translation. If this translate function is in use it is harder to implement it later.

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/resources/xsl/functions/i18n.xsl (hunk 1 above)
Comment (sebhofmann) [reply mycore-base/src/main/resources/xsl/functions/i18n.xsl:12]: For the replacements?

**Comment by yagee-de** [comment]
Added support for arguments:
```xsl
<xsl:value-of select="mcri18n:translate-with-params('mir.pagination.entriesInfo', ('foo', 'bar', xs:string(fn:current-time())))" />
```
Without any arguments:
```xsl
<xsl:value-of select="mcri18n:translate('ACLE.alert.access.add.error')" />
```

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 419a49b8e74f5eb44c9d8a5b8db9bd19e917e55e! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6048-pull_request-1066

**Review by yagee-de — APPROVED** [review]

**Review by kkrebs — APPROVED** [review]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on c7bfa4f3569cd8bea5ec9e937391e46f04062ddd! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6058-pull_request-1066

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on c7bfa4f3569cd8bea5ec9e937391e46f04062ddd! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6058-pull_request-1066

**Pull request merged**